README.md

```diff
@@ -1,20 +1,157 @@
+##### `VERSION <string>`
+The version of the application.
+##### `DISPLAY_NAME <string>`
+The name displayed to users. Defaults to NAME.
+##### `PUBLISHER_DISPLAY_NAME <string>`
+The name of the publisher.
+##### `IDENTIFIER <string>`
+A unique bundle identifier for the application.
+##### `CATEGORY <string>` 
+Category for the app in the macOS App Store or Finder.
+##### `TARGET <target>`
+An existing CMake target representing the main application executable.
+##### `EXECUTABLE <path>`
+Path to the application executable (if not providing the TARGET option).
+
+#### `cmakeadd_macos_bundle`
+The core function to create a complete macOS application bundle (.app)
+```cmake
+cmakeadd_macos_bundle(
+    <target> 
+    DESTINATION <path> 
+    [INFO <path>] 
+    [ICON <path>] 
+    [TARGET <target>] 
+    [EXECUTABLE <path>] 
+    [RESOURCES [FILE|DIR <from> <to>]...] 
+    [DEPENDS <target...>])
+```
+#### Options
+##### `<target>`
+The name of the primary CMake target for the macOS application. This is used to link dependencies and create the overall bundle structure.
+##### `DESTINATION <path>`
+The desired output path for the generated application bundle. Defaults to <target_name>.app in the build directory.
+##### `INFO <path>`
+Path to a pre-existing Info.plist file. If not provided, one will be generated automatically.
+##### `ICON <path>`
+Path to a .icns file containing the application's icon.
+##### `TARGET <target>`
+Name of a CMake target representing the core executable of the application.
+##### `EXECUTABLE <path>`Direct path to the application executable (use if not providing TARGET).
+##### `RESOURCES [FILE|DIR <from> <to>] ...`
+A list of additional resources to include in the app bundle: 
+* FILE <from> <to>: Copies a single file from <from> to <to> location within the bundle's "Resources" directory.
+* DIR <from> <to>: Copies an entire directory from <from> to <to> within the bundle's "Resources" directory.
+##### `DEPENDS <target...>` 
+A list of CMake targets on which the bundle creation process depends.
+
+#### `code_sign_macos`
+Code signs a macOS application or bundle, enhancing security.
+```cmake
+code_sign_macos(
+    <target> 
+    [PATH <path>] 
+    [TARGET <target>] 
+    [ENTITLEMENTS <path>] 
+    IDENTITY <string> 
+    [KEYCHAIN <string>] 
+    [DEPENDS <target...>]
+)
+```
+#### Options
+##### `<target>`
+The name of a CMake target to which the code signing will be applied.
+##### `PATH <path>`
+The path to the macOS application bundle or executable to be signed.
+##### `TARGET <target>`
+A CMake target representing the application to be signed (alternative to providing PATH).
+##### `ENTITLEMENTS <path>`
+The path to an Entitlements.plist file specifying permissions for the application.
+##### `IDENTITY <string>`
+The code signing identity to use (from the developer certificate).
+##### `KEYCHAIN <string>`
+The path to the keychain containing the code signing certificate and private key.
+
```

Comment:
````suggestion
#### `add_macos_bundle_info`
Generates the `Info.plist` file, essential for macOS application metadata.

```cmake
add_macos_bundle_info(
  <target>
  [DESTINATION <path>]
  NAME <string>
  VERSION <string>
  DISPLAY_NAME <string>
  PUBLISHER_DISPLAY_NAME <string>
  IDENTIFIER <identifier>
  CATEGORY <string>
  [TARGET <target>]
  [EXECUTABLE <path>]
)
```

##### `<target>`
The name of the CMake target to create.

##### `DESTINATION <path>`
Specifies the location to save the `Info.plist` file. Defaults to `"Info.plist"` in the build directory.

##### `NAME <string>`
The name of the application.

##### `VERSION <string>`
The semantic version of the application.

##### `DISPLAY_NAME <string>`
The name displayed to users. Defaults to `"${NAME}"`.

##### `PUBLISHER_DISPLAY_NAME <string>`
The name of the publisher.

##### `IDENTIFIER <string>`
A unique bundle identifier for the application.

##### `CATEGORY <string>` 
Category for the app in the macOS App Store or Finder.

##### `TARGET <target>`
An existing CMake target representing the main application executable.

##### `EXECUTABLE <path>`
Path to the application executable. Use if not providing the `TARGET` option.

#### `add_macos_bundle`
The core function to create a complete macOS application bundle (`.app` file).

```cmake
cmakeadd_macos_bundle(
  <target> 
  DESTINATION <path>
  [INFO <path>]
  [ICON <path>]
  [TARGET <target>]
  [EXECUTABLE <path>]
  [RESOURCES [FILE|DIR <from> <to>]...]
  [DEPENDS <target...>]
)
```

##### `<target>`
The name of the CMake target to create for the macOS application.

##### `DESTINATION <path>`
The desired output path for the generated application bundle. Defaults to `"${NAME}.app` in the build directory.

##### `INFO <path>`
Path to an `Info.plist` file.

##### `ICON <path>`
Path to a `.icns` file containing the application icon.

##### `TARGET <target>`
Name of a CMake target representing the core executable of the application.

##### `EXECUTABLE <path>`
Direct path to the application executable. Use if not providing `TARGET`.

##### `RESOURCES [FILE|DIR <from> <to>] ...`
A list of additional resources to include in the app bundle.

###### `FILE <from> <to>`
Copies a single file from `<from>` to `<to>` location within the `Resources` directory of the bundle.

###### `DIR <from> <to>`
Copies an entire directory from `<from>` to `<to>` within the `Resources` directory of the bundle.

##### `DEPENDS <target...>` 
A list of CMake targets on which the bundle creation process depends.

#### `code_sign_macos`
Code signs a macOS application or bundle, enhancing security.

```cmake
code_sign_macos(
  <target> 
  [PATH <path>] 
  [TARGET <target>] 
  [ENTITLEMENTS <path>] 
  IDENTITY <string> 
  [KEYCHAIN <string>] 
  [DEPENDS <target...>]
)
```

##### `<target>`
The name of the CMake target to create.

##### `PATH <path>`
The path to the macOS application bundle or executable to be signed.

##### `TARGET <target>`
A CMake target representing the application to be signed. Alternative to providing `PATH`.

##### `ENTITLEMENTS <path>`
The path to an `Entitlements.plist` file specifying permissions for the application.

##### `IDENTITY <string>`
The code signing identity to use.

##### `KEYCHAIN <string>`
The path to the keychain containing the code signing certificate and private key.
````